src/main/java/com/todoapp/app/restcontroller/TaskResource.java

```diff
@@ -0,0 +1,49 @@
+package com.todoapp.app.restcontroller;
+
+import com.todoapp.app.entity.Task;
+import com.todoapp.app.repository.Task_Repository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class TaskResource {
+
+    @Autowired
+    Task_Repository task_repository;
+
+    @GetMapping( "/users/{user_id}/pending_tasks" )
+    public Optional<List<Task>> get_pending_tasks(@PathVariable int id ){
+
+        return task_repository.findByTidAndTask_status( id , "Pending" );
+    }
+
+    @PostMapping( "/users/{user_id}/tasks" )
+    public void add_task(@PathVariable int id , @RequestBody Task task ){
+
+        task_repository.save( task );
+    }
+
+    @DeleteMapping( "/users/{user_id}/task/{task_id}" )
+    public void delete_pending_task( @PathVariable int user_id , @PathVariable int task_id ){
+
```

Comment:
How you are making sure that this task is related to this provided user not other user ? What if I try to delete a task by giving a task id but it belongs to other users ?

Reply:
how an user will know the task id of tasks of other users.  And task_id is an primary key which will be unique in task table.

Reply:
maaf karde yaar. Galtii hoo gayiiii